### ISSUE TITLE
Helm chart should use StorageClassName

### ISSUE THREAD

**Issue by agrahamlincoln** [fission/fission#444]
I deployed fission on K8S 1.8.4 using the helm chart and the fission-storage-pvc was not fulfilled.
I looked at the yaml and noted that the chart is still using the deprecated storage-class annotations:
```
$ kc get pvc -n fission -o yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: PersistentVolumeClaim
  metadata:
    annotations:
      volume.alpha.kubernetes.io/storage-class: default
    creationTimestamp: 2018-01-11T17:23:40Z
    labels:
      app: fission-storage
      chart: fission-all-0.4.0
      release: fission
    name: fission-storage-pvc
    namespace: fission
    resourceVersion: "57328"
    selfLink: /api/v1/namespaces/fission/persistentvolumeclaims/fission-storage-pvc
    uid: 2ab82f10-f6f4-11e7-b70a-fa163e22de15
  spec:
    accessModes:
    - ReadWriteOnce
    resources:
      requests:
        storage: 8Gi
  status:
    phase: Pending
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

See https://github.com/kubernetes/charts/issues/1869 for examples of other helm charts implementing this.

**Comment by agrahamlincoln** [comment]
To add some context to this. StorageClassName was added in K8S 1.6. (See https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.6.md#volumes-2)
The volume.alpha.kubernetes.io/storage-class annotation is removed in 1.8
The volume.beta.kubernetes.io/storage-class annotation still works in 1.9, but will be removed in a future release. (See https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md#storage-2)

**Comment by agrahamlincoln** [comment]
Closed by #445
